Homework29

rramoscode opened the pull request.

rramoscode committed:
Homework16 Test Registration Navigation

-Create a new file, Homework16.java in IntelliJ IDEA.
-Create a test case, registrationNavigation() using @Test annotation.
-Using Selenium, navigate to "https://bbb.testpro.io/".
-Click the Registration link.
-Verify that you are redirected to the Registration page using the Assert method.

rramoscode committed:
Test Case  ( addSongToPlaylist()  ) using @Test annotation and the helper/reusable methods we created.

steps completed:
- Navigate to "https://bbb.testpro.io/".
- Log in with your credentials.
-  Search for a song (choose any song of your choice).
- Click View All to display the search results.
- Click the first song in the search results.
- Click ADD TO...
- Choose the playlist to add it to, (we can create a new one with a unique name).
- Verify that the notification message appears.
- Verify that notification message has the text, "Added 1 song into {your playlist}".

rramoscode committed:
Update BaseTest.java

updated note for personal reference

rramoscode committed:
Homework18 created Tc for playSong and new .xml file with method to run

- Create a test case, playSong() using @Test annotation.
- Use the helper/reusable methods we created.
- Create a new config XML file for the test case.
- Specify the method to be ran in the XML file.
- Navigate to "https://bbb.testpro.io/".
- Log in with your credentials.
- Click «Play next song» (media player controls), then the Play button, to play a song.
- Validate that a song is playing by verifying if the sound bar is displayed.

rramoscode committed:
testing no changes made

review code option no real changes made to of previous code

rramoscode committed:
Parameter Annotation in TestNG

 - Navigate to "https://bbb.testpro.io/".
- Log in with your credentials.
- Select the playlist you want to delete.
- Click the red button "x PLAYLIST" on the top right part of the page, and click on it.
- Verify that the confirmation notification displayed has the text, "Deleted playlist  ".

rramoscode committed:
no changes made

there was no changes made , was looking at different option for waits

rramoscode committed:
refactored code to have Explicit Wait

refactored code to have Explicit Waits, instead for Thread.sleep.
- Refactor your code - implement Explicit waits.
- Remove all implementations of Thread.sleep.
- Make sure our codes run without errors.

rramoscode committed:
Explicit wait added to all methods

Added explicit wait to all methods for HW 16-18 although not all methods may be needed for HW20.

rramoscode committed:
recommended updates made

removed   throws interruptedExceptions  and unused method for non-empty playlist

rramoscode committed:
rename playlist code

- Create a new file Homework21.java for the renamePlaylist script
- Used Action Class methods, create/refactor the test case renamePlaylist().
- Implemented Explicit Waits.
- Remove all implementations of Thread.sleep. (used only during debugging)

script changes rrPlaylist3 to rrPlaylist4

rramoscode committed:
refactor Pull member

updated Homework21 and refactored helper methods using pull member to move helper methods to BaseTest

rramoscode committed:
Update BaseTest.java

update made due to noticed the helper methods had protected void... instead of public void

rramoscode committed:
Update file-system.probe

reran Homework21 no issues

rramoscode committed:
update to Assert

updated assertion to include string name

rramoscode committed:
update to Assertion

practice change for different looks

rramoscode committed:
homework22 refactor and implement code to POM

Refactored the code, implemented and created Page Objects.

rramoscode committed:
Update BasePage.java

per feedback
updated variable to protected to prevent non-related classes from trying to use it.

rramoscode committed:
Update LoginPage.java

implemented encapsulation to the Loginpage

rramoscode committed:
Homework23 selenium page Factory

- Refactored the code and implemented Page Factory by using the initElements() method of Page Factory class.
-  Used @FindBy annotation for declaring Web Elements.

rramoscode committed:
Update file-system.probe

no change made

rramoscode committed:
Update file-system.probe

no changes made

rramoscode committed:
suggested changes made

made suggested changes based on feedback

rramoscode committed:
Update BaseTest.java

rramoscode committed:
homework24 using standalone option locally

- Set up  Selenium Grid (Standalone option) locally. Chrome/ firefox browser etc..
- Implement Browser Factory method as we had in the class, so I could run Chrome browser using your local Grid.

test changes rrPlaylist3 to rrPlaylist4

rramoscode committed:
Update BaseTest.java

updated  edge information (educational purposes only)

rramoscode committed:
Update BaseTest.java

no changes to code made

rramoscode committed:
Homework25 cloud execution

- created new branch Homework25
- added the cloud execution capability to the framework for use with  Lamdatest.com
- ran @test renamePlaylist in the cloud

rramoscode committed:
aded to to TestNG

added additional @test to TestNG and ran a test through cloud LambdaTest  all passed.
- LoginTest - sucessfulLonginTest - Passed
- Homework20 removeplaylist (rrPlaylist2) - Passed
-Homework24 renamePlaylist - renames rrPlaylist3 to rrPlaylist4 - Passed

rramoscode committed:
update to correct errors

test working when using - gradle clean test - will resubmit for review
test performed:
successful login test
delete rrPlaylist2
rename rrPlaylist3 to rrPlaylist4

rramoscode committed:
Cucumber for TestNg

installed Cucumber for TestNG and configured it to run a login Testcase, successful login.

gradle clean Logintests

rramoscode committed:
Update file-system.probe

rramoscode committed:
removed and updated file not relevant to Smoke and Regression

created regression and smoke test .xml files and included relevant tests for each
add song to play list rrPlaylist
remove playlist rrPlaylist5
change playlist name rrPlaylist3 to rrPlaylist4

rramoscode committed:
updated code and created  Smoke and Regression xml.

1. Create a new branch named Homework29 using GitHub Desktop.
2. Remove the Homework.java classes that are irrelevant to the Smoke & Regression tests.
3. Move Login test cases to LoginTests.java .

Ran Smoke and Regression test and all passed.
gradle clean testSmoke
gradle clean testRegression

Berichicko commented:
Hello,

I have checked your homework 29, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)

rramoscode committed:
added method

Added to LoginTests to test for empty password field using method LoginEmptyPasswordTest

rramoscode committed:
update to test pro URL

a change was made to URL for Koel, updated to new URL qa.koel.app
